stl/inc/memory

```diff
@@ -3079,15 +3079,21 @@ inline void* align(size_t _Bound, size_t _Size, void*& _Ptr, size_t& _Space) noe
     return _Ptr;
 }
 
-#if _HAS_CXX20 && 0
+#if _HAS_CXX20
 template <size_t _Nx, class _Ty>
 _NODISCARD constexpr _Ty* assume_aligned(_Ty* _Ptr) noexcept /* strengthened */ {
-    // this enforces the requirement that _Nx be a power of two
-    __builtin_assume_aligned(_Ptr, _Nx);
+#if 1 // TRANSITION: delete after __builtin_assume_aligned will be constexpr

```

Comment:
What compilers (MSVC, Clang, EDG IntelliSense) don't support `constexpr __builtin_assume_aligned`? We need to file bug reports (on DevCom and LLVM) if we want this to be fixed, then we need to cite the bug numbers in this comment.

Reply:
Sorry: this was my fault. I suggested the `is_constant_evaluated` hack to unblock this work while I investigate what regressed the `constexpr` intrinsic in MSVC. (It's been so long since the compiler backend team implemented this I started doubting my memory that the intrinsic was working in `constexpr`.) I determined that it definitely _should_ be working but is not - some `constexpr` change has regressed it.

@fsb4000 do you mind filing a DevCom bug noting that neither MSVC nor EDG/IntelliSense accept this intrinsic in `constexpr`? I could take care of it, otherwise.

Reply:
@CaseyCarter I can create issue. I have test case which works for clang-cl and doesn't work for MSVC.
How do you check EDG/IntelliSense?

Reply:
You can take an MSVC command line and add `/c /BE` to say "compile-only (don't link) with the secret undocumented option that activates the EDG front-end".